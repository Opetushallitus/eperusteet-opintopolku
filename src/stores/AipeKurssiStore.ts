import  { reactive, computed } from'vue';
import _ from 'lodash';
import { Aipeopetuksensisalto, AIPEOppiaineDto, AIPEKurssiDto } from '@shared/api/eperusteet';

export class AipeKurssiStore {
  private state = reactive({
    oppiaine: null as AIPEOppiaineDto | null,
    kurssi: null as AIPEKurssiDto | null,
  });

  public static async create(perusteId: number, vaiheId: number, oppiaineId: number, kurssiId: number) {
    const result = new AipeKurssiStore(perusteId, vaiheId, oppiaineId, kurssiId);
    await result.init();
    return result;
  }

  constructor(private perusteId: number, private vaiheId: number, private oppiaineId: number, private kurssiId: number) {
  }

  public readonly kurssi = computed(() => this.state.kurssi);
  public readonly oppiaine = computed(() => this.state.oppiaine);

  public async init() {
    this.state.kurssi = (await Aipeopetuksensisalto.getAipeKurssi(this.perusteId, this.vaiheId, this.oppiaineId, this.kurssiId)).data;
    this.state.oppiaine = (await Aipeopetuksensisalto.getAipeOppiaine(this.perusteId, this.vaiheId, this.oppiaineId)).data;
  }
}
