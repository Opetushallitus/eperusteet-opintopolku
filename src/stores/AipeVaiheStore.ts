import  { reactive, computed } from'vue';
import _ from 'lodash';
import { Aipeopetuksensisalto, AIPEVaiheDto } from '@shared/api/eperusteet';

export class AipeVaiheStore {
  private state = reactive({
    vaihe: null as AIPEVaiheDto | null,
  });

  public static async create(perusteId: number, vaiheId: number) {
    const result = new AipeVaiheStore(perusteId, vaiheId);
    await result.init();
    return result;
  }

  constructor(private perusteId: number, private vaiheId: number) {
  }

  public readonly vaihe = computed(() => this.state.vaihe);

  public async init() {
    this.state.vaihe = (await Aipeopetuksensisalto.getVaihe(this.perusteId, this.vaiheId)).data;
  }
}
