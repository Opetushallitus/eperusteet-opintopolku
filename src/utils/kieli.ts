export function highlight(nimi, query: string) {
  if (!query) {
    return nimi;
  }

  return (nimi || '').replace(new RegExp(query, 'ig'), (match) => '<mark>' + match + '</mark>');
}
