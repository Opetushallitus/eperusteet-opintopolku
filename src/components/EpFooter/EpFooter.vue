<template>
  <div class="container">
    <footer>
      <div class="horizontal-line">
        <router-link :to="{ name: 'root'}">
          <img
            :src="ophLogo"
            :alt="$t('eperusteet')"
          >
        </router-link>
      </div>
      <div class="footer-row">
        <div class="footer-content">
          {{ $t('footer-eperusteet-palvelu') }}
        </div>
        <div class="footer-links">
          <EpPalauteLinkki yllapito-avain="eperusteet-opintopolku-palaute-url" />
          <div class="d-flex">
            <EpMaterialIcon>chevron_right</EpMaterialIcon>
            <ep-linkki :url="$kaanna(linkit.saavutettavuusseloste)">
              {{ $t('saavutettavuusseloste') }}
            </ep-linkki>
          </div>
          <div class="d-flex">
            <EpMaterialIcon>chevron_right</EpMaterialIcon>
            <EpLinkki
              :url="$kaanna(linkit.virkailija)"
              icon="launch"
              icon-right
            >
              {{ $t('siirry-virkailijanakymaan') }}
            </EpLinkki>
          </div>
        </div>
      </div>
      <div class="horizontal-line">
        <a :href="$kaanna(linkit.oph)">
          <img
            class="logo"
            :src="opintopolkuLogo"
            :alt="$t('oph')"
          >
        </a>
      </div>
      <div class="footer-row">
        <div class="footer-oph">
          {{ $t('footer-oph') }}
        </div>
      </div>
    </footer>
  </div>
</template>

<script setup lang="ts">
import { $kaanna } from '@shared/utils/globals';
import EpLinkki from '@shared/components/EpLinkki/EpLinkki.vue';
import ophLogo from '@assets/img/banners/opintopolku/logo.svg';
import opintopolkuLogo from '@assets/img/banners/opintopolku/oph_logo2.png';
import EpPalauteLinkki from '@shared/components/EpPalauteLinkki/EpPalauteLinkki.vue';
import EpMaterialIcon from '@shared/components/EpMaterialIcon/EpMaterialIcon.vue';

const linkit = {
  oph: {
    fi: 'https://www.oph.fi/fi',
    sv: 'https://www.oph.fi/sv',
    en: 'https://www.oph.fi/en',
  },
  virkailija: {
    fi: 'http://virkailija.opintopolku.fi',
    sv: 'http://virkailija.opintopolku.fi',
  },
  palaute: {
    fi: 'mailto:eperusteet@opintopolku.fi',
    sv: 'mailto:eperusteet@opintopolku.fi',
  },
  saavutettavuusseloste: {
    fi: 'https://opintopolku.fi/konfo/fi/sivu/eperusteet-saavutettavuusseloste',
    sv: 'https://opintopolku.fi/konfo/sv/sivu/tillganglighetsutlatande-for-egrunder',
  },
};
</script>

<style scoped lang="scss">
@import '@shared/styles/_variables.scss';

footer {
  padding-bottom: 50px;
  hyphens: none;
  margin-top: 100px;

  .horizontal-line {
    margin-top: 70px;
    width: 100%;
    border: none;
    padding: 0;
    overflow: visible;
    text-align: center;
    background-color: #FFF;
    border-top-color: $gray-lighten-3;
    border-top-style: solid;
    border-top-width: 1px;

    img {
      top: -30px;
      padding: 0 50px;
      position: relative;
      background-color: #FFF;
    }
  }

  .footer-content {
    width: 65%;
    display: flex;
    margin: 0 auto;

    @media (max-width: 991.98px) {
      padding: 0 20px;
      width: 100%;
    }
  }

  .footer-links {
    width: 35%;
    padding-left: 20px;
    color: $oph-green;

    @media (max-width: 991.98px) {
      padding: 20px;
      width: 100%;
    }
  }

  .footer-oph {
    justify-content: unset;
    font-size: 0.875rem;
    margin: 0 auto;
    width: 60%;

    @media (max-width: 991.98px) {
      padding: 0 20px;
      width: 100%;
    }
  }

  .footer-row {
    display: flex;

    @media (max-width: 991.98px) {
      display: grid;
    }
  }
}

.logo {
  width: 200px;
}

:deep(.linkki) {
  font-size: 1rem;
  padding-bottom: 3px;

  a {
    color: $oph-green;
  }
}
</style>
