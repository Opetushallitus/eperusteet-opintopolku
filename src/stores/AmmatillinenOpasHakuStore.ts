import { Store, Getter, State } from '@shared/stores/store';
import { PerusteenJulkaisuData } from '@shared/api/eperusteet';
import { Kielet } from '@shared/stores/kieli';
import { julkaistutPerusteet, JulkaistutPerusteetQuery } from '@/api/eperusteet';
import _ from 'lodash';
import { IPerusteHakuStore } from './IPerusteHakuStore';
import { Debounced } from '@shared/utils/delay';

@Store
export class AmmatillinenOpasHakuStore implements IPerusteHakuStore {
  @State() public perusteet: PerusteenJulkaisuData[] | null = null;
  @State() public page = 0;
  @State() public pages = 0;
  @State() public total = 0;
  @State() public perPage = 10;

  @State()
  public filterdata: JulkaistutPerusteetQuery = {
    nimi: '',
    koulutustyyppi: [
      'koulutustyyppi_1',
      'koulutustyyppi_11',
      'koulutustyyppi_12',
      'koulutustyyppi_5',
      'koulutustyyppi_18',
    ],
    tuleva: true,
    voimassaolo: true,
    siirtyma: false,
    poistunut: false,
    tyyppi: 'opas',
  };

  constructor(data: JulkaistutPerusteetQuery = { }) {
    this.filterdata = {
      ...this.filterdata,
      ...data,
    };
  }

  @Getter((state) => {
    return [
      'tuleva',
      'voimassaolo',
    ];
  })
  public readonly toggles!: string[];

  @Getter((state) => ({
    ...state.filterdata,
    sivu: state.page,
    sivukoko: state.perPage,
    kieli: Kielet.getSisaltoKieli.value,
  }))
  public readonly filters!: JulkaistutPerusteetQuery;

  @Debounced(500)
  async fetch() {
    this.perusteet = null;
    const result = await julkaistutPerusteet(this.filters);
    this.total = result['kokonaismäärä'];
    this.page = result.sivu;
    this.perPage = result.sivukoko;
    this.pages = result.sivuja;
    this.perusteet = result.data;
  }

  async updateFilters(filters: JulkaistutPerusteetQuery) {
    this.filterdata = {
      ...this.filters,
      ...filters,
    };
    await this.fetch();
  }
}
