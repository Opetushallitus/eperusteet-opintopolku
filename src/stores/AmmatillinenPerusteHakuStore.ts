import { defineStore } from 'pinia';
import { Arviointiasteikot, PerusteenJulkaisuData } from '@shared/api/eperusteet';
import { Kielet } from '@shared/stores/kieli';
import { julkaistutPerusteet, JulkaistutPerusteetQuery } from '@/api/eperusteet';
import { ref, computed } from 'vue';
import { debounced, DEFAULT_PUBLIC_WAIT_TIME_MS } from '@shared/utils/delay';
import _ from 'lodash';
import { Page } from '@shared/tyypit';

export const useAmmatillinenPerusteHakuStore = defineStore('ammatillinenPerusteHaku', () => {
  // State
  const perusteet = ref<Page<PerusteenJulkaisuData> | null>(null);
  const arviointiasteikot = ref<any[]>([]);

  const filterToQueryParams = {
    nimiTaiKoodi: 'haku',
    tutkintotyyppi: 'tutkintotyyppi',
    tuleva: 'tuleva',
    voimassaolo: 'voimassaolo',
    siirtyma: 'siirtyma',
    poistunut: 'poistunut',
    perusteet: 'perusteet',
    tutkinnonosat: 'tutkinnonosat',
    sivu: 'sivu',
  };

  const fetch = debounced(async (filters: JulkaistutPerusteetQuery) => {
    perusteet.value = null;
    perusteet.value = await julkaistutPerusteet(
      {
        ...filters,
        sivukoko: 10,
      },
    );
  });

  async function fetchArviointiasteikot() {
    arviointiasteikot.value = (await Arviointiasteikot.getAll()).data;
  }

  return {
    // State
    perusteet,
    arviointiasteikot,

    filterToQueryParams,

    // Actions
    fetch,
    fetchArviointiasteikot,
  };
});
