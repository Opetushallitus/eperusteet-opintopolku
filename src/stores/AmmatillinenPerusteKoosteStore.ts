import Vue from 'vue';
import VueCompositionApi, { reactive, computed, ref, watch } from '@vue/composition-api';
import _ from 'lodash';
import { OpetussuunnitelmaDto, getJulkisetOpetussuunnitelmat, OpetussuunnitelmaQuery } from '@shared/api/amosaa';
import { TiedoteDto, Perusteet, PerusteKaikkiDto } from '@shared/api/eperusteet';
import { Kielet } from '@shared/stores/kieli';
import { tiedoteQuery } from '@/api/eperusteet';
import { Page } from '@shared/tyypit';
import { Debounced } from '@shared/utils/delay';
import { AmmatillisetKoulutustyypit } from '@shared/utils/perusteet';
import { usePerusteCacheStore } from '@/stores/PerusteCacheStore';

Vue.use(VueCompositionApi);

export class AmmatillinenPerusteKoosteStore {
  public state = reactive({
    peruste: null as PerusteKaikkiDto | null,
    opetussuunnitelmat: null as Page<OpetussuunnitelmaDto> | null,
    tiedotteet: null as TiedoteDto | null,
    opsQuery: null as OpetussuunnitelmaQuery | null,
  });

  constructor(private perusteId: number) {
    this.fetch();
  }

  public readonly peruste = computed(() => this.state.peruste);
  public readonly tiedotteet = computed(() => this.state.tiedotteet);
  public readonly opetussuunnitelmat = computed(() => this.state.opetussuunnitelmat);
  public readonly perusteCacheStore = usePerusteCacheStore();

  public async fetch() {
    this.state.peruste = null;
    this.perusteCacheStore.addPerusteStore(this.perusteId);
    this.state.peruste = (await Perusteet.getKokoSisalto(this.perusteId)).data;

    this.state.opsQuery = {
      perusteenDiaarinumero: this.state.peruste!.diaarinumero,
      perusteId: this.state.peruste!.id,
      sivu: 0,
      sivukoko: 10,
      kieli: Kielet.getUiKieli.value,
      koulutustyyppi: AmmatillisetKoulutustyypit,
    };

    await this.fetchOpetussuunnitelmat(this.state.opsQuery);

    const vanhat = (await tiedoteQuery({
      sivukoko: 100,
      perusteId: this.perusteId,
      julkinen: true,
    }));

    const uudet = (await tiedoteQuery({
      sivukoko: 100,
      perusteIds: [this.perusteId],
    }));

    this.state.tiedotteet = _.chain([
      ...vanhat,
      ...uudet,
    ])
      .filter(tiedote => !!tiedote.otsikko)
      .uniqBy('id')
      .sortBy('luotu')
      .reverse()
      .value() as any;
  }

  @Debounced(300)
  public async fetchOpetussuunnitelmat(query) {
    this.state.opetussuunnitelmat = null;
    this.state.opetussuunnitelmat = ((await getJulkisetOpetussuunnitelmat({ ...this.state.opsQuery, ...query })).data as any);
  }
}
