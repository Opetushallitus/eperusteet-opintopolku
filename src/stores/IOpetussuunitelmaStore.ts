import { NavigationNode, NavigationFilter } from '@shared/utils/NavigationBuilder';
import { Cancelable } from 'lodash';

export interface IOpetussuunnitelmaStore {
  opetussuunnitelma: any;
  tila: any,
  koulutustyyppi: string;
  current: NavigationNode | null;
  sidenavFilter: NavigationFilter;
  filteredSidenav: NavigationNode | null;
  flattenedSidenav: NavigationNode[];
  sidenavLoading: boolean;
  updateRoute: (options) => any;
  updateFilter: ((filter: NavigationFilter) => void) & Cancelable;
};
