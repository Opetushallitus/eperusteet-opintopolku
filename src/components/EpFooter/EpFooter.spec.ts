import { mount } from '@vue/test-utils';
import EpFooter from './EpFooter.vue';
import { Kielet } from '@shared/stores/kieli';
import { mocks, stubs } from '@shared/utils/jestutils';
import VueI18n from 'vue-i18n';
import { Kaannos } from '@shared/plugins/kaannos';
import { nextTick } from 'vue';
import { globalStubs } from '@shared/utils/__tests__/stubs';

describe('EpFooter', () => {
  test('Renders', async () => {
    const wrapper = mount(EpFooter as any, {
      global: {
        ...globalStubs,
      },
    });

    nextTick();
    expect(wrapper.html()).toMatchSnapshot();
  });
});
