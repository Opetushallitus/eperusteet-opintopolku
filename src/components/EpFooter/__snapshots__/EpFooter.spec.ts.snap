// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EpFooter Renders 1`] = `
<div class="container">
  <footer>
    <div class="horizontal-line"><routerlink-stub to="[object Object]"><img src="" alt="eperusteet"></routerlink-stub></div>
    <div class="footer-row">
      <div class="footer-content">
        footer-eperusteet-palvelu
      </div>
      <div class="footer-links">
        <div class="d-flex">
          <epmaterialicon>chevron_right</epmaterialicon> <span>palaute:</span>
          <div class="linkki ml-1"><a href="mailto:eperusteet@opintopolku.fi" rel="noopener noreferrer" target="_blank"><!----> <span>eperusteet@opintopolku.fi</span></a></div>
        </div>
        <div class="d-flex">
          <epmaterialicon>chevron_right</epmaterialicon>
          <div class="linkki"><a href="https://opintopolku.fi/konfo/fi/sivu/eperusteet-saavutettavuusseloste" rel="noopener noreferrer" target="_blank"><!----> saavutettavuusseloste</a></div>
        </div>
        <div class="d-flex">
          <epmaterialicon>chevron_right</epmaterialicon>
          <div class="linkki"><a href="http://virkailija.opintopolku.fi" rel="noopener noreferrer" target="_blank"><!----> siirry-virkailijanakymaan</a></div>
          <epmaterialicon size="20px" class="ml-1">launch</epmaterialicon>
        </div>
      </div>
    </div>
    <div class="horizontal-line"><a href="https://www.oph.fi/fi"><img src="" alt="oph" class="logo"></a></div>
    <div class="footer-row">
      <div class="footer-oph">
        footer-oph
      </div>
    </div>
  </footer>
</div>
`;
